### PR TITLE
selinux: Update to work properly with el8toel9 upgrade path

### PR DESCRIPTION
**Pull request by vmojzis**
- Add rhel8 specific list of "removed types"
- Improve handling of modules shipped as DSP
- Update install-deps
- Improve regexp for matching "semodule -lfull" entries
- Add compatibility policy module for running tests on newer systems
- Retry policy module removal in case of failure (inter-module dependency)
    
Signed-off-by: Vit Mojzis <vmojzis@redhat.com>

### CODE REVIEW

**Comment by vmojzis** [comment]
Ready for review.
I'd still like to add some kind of reporting functionality (list modules and other customisation that where transferred, point out any issues). Could you please point me to some documentation for this soft of thing (or maybe an example of an actor that utilises it)?

**Comment by pirat89** [comment]
@vmojzis thanks! we will look at it probably after the upcoming release is out. For the reporting I suggest this doc:
* https://leapp.readthedocs.io/en/latest/el7toel8/actor-rhel7-to-rhel8.html#producing-data-for-other-actors-and-reporting

File: repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/mock_modules/mock2.cil [hunk 1]
```diff
@@ -1,4 +1,6 @@
 (type mock_type2_t)
 (typeattributeset direct_run_init (mock_type2_t))
 (allow mock_type_t file_type (file (getattr open read)))
+(optional mock2_optional
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/mock_modules/mock2.cil:4]: Isn't formatting a bit off here?
```
(optional mock2_optional
   (allow mock_type_t ganesha_exec_t (file (getattr open read)))
)
```

File: repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py [hunk 2]
```diff
@@ -2,10 +2,10 @@
 import shutil
 
 from leapp.actors import Actor
-from leapp.models import SELinuxModules, SELinuxCustom, SELinuxRequestRPMs
+from leapp.models import SELinuxModules, SELinuxCustom
 from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
 from leapp.libraries.stdlib import run, CalledProcessError
-
+from leapp.libraries.actor import selinuxapplycustom
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:8]: Hmm wondering why didn't the linter catch this. Could you please swap L7 and L8? And I'll check how is our linter doing in the meantime)

File: repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/tests/mock_modules/mock2.cil [hunk 3]
```diff
@@ -1,4 +1,6 @@
 (type mock_type2_t)
 (typeattributeset direct_run_init (mock_type2_t))
 (allow mock_type_t file_type (file (getattr open read)))
+(optional mock2_optional
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/tests/mock_modules/mock2.cil:4]: Isn't formatting off here?

**Review by fernflower — COMMENTED** [review]
Couple of nits.

**Comment by vmojzis** [comment]
@fernflower Thanks, I'll fix that. Meanwhile I've been working on the reporting and it occurred to me that it could be useful to save any policy modules that failed to be applied on the upgraded system. Is there some path that leapp uses for such things, or can I just use something like /tmp/selinux-leapp (printing the modules as part of the report is not an option since they can be quite long)?

**Comment by vmojzis** [comment]
@fernflower Sorry for the delay. Ready for another review. I added some new functionality and improved tests to cover it as well. Still not sure about the backup directory (see https://github.com/oamg/leapp-repository/pull/749#issuecomment-963281031)

**Comment by fernflower** [comment]
@vmojzis Could you please rebase against master? We have tuned lint target to catch isort violations and introduced a lint_fix target that should auto fix them.

**Comment by vmojzis** [comment]
@fernflower Rebased. But for some reason `make fast_lint` stopped working for me.
Before the rebase I got 
`# make fast_lint                                                                                                                                      130 ↵
Using config file /home/vmojzis/devel/leapp/leapp-repository/.pylintrc

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

Using config file /home/vmojzis/devel/leapp/leapp-repository/.pylintrc

------------------------------------
Your code has been rated at 10.00/10`

But now I get:

`# make fast_lint                                                  

No files to lint.
`

**Comment by fernflower** [comment]
@vmojzis What is the top commit hash of your local master?

File: repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py (hunk 2 above)
Comment (fernflower) [reply repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:8]: lint_fix make target should help with that. This relays on local master for comparison, so please make sure that not only your pr is rebased against master, but master is also up to date.

Comment (fernflower) [reply repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:8]: Or you can apply these changes
```
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py b/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py
index e0ae41b..7eca815 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py
@@ -3,11 +3,11 @@ import shutil
 
 from leapp import reporting
 from leapp.actors import Actor
-from leapp.models import SELinuxModules, SELinuxCustom
-from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
 from leapp.libraries.actor import selinuxapplycustom
 from leapp.libraries.actor.selinuxapplycustom import BACKUP_DIRECTORY
-from leapp.libraries.stdlib import run, CalledProcessError
+from leapp.libraries.stdlib import CalledProcessError, run
+from leapp.models import SELinuxCustom, SELinuxModules
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
 
 WORKING_DIRECTORY = '/tmp/selinux/'
 
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/libraries/selinuxapplycustom.py b/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/libraries/selinuxapplycustom.py
index 95c06bf..887f381 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/libraries/selinuxapplycustom.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/libraries/selinuxapplycustom.py
@@ -1,9 +1,8 @@
-import re
 import os
+import re
 import shutil
 
-from leapp.libraries.stdlib import api, run, CalledProcessError
-
+from leapp.libraries.stdlib import api, CalledProcessError, run
 
 BACKUP_DIRECTORY = "/tmp/leapp-selinux/"
 
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/tests/component_test_selinuxapplycustom.py b/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/tests/component_test_selinuxapplycustom.py
index b0ed347..0171f58 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/tests/component_test_selinuxapplycustom.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxapplycustom/tests/component_test_selinuxapplycustom.py
@@ -2,8 +2,8 @@ import os
 
 import pytest
 
-from leapp.models import SELinuxModule, SELinuxModules, SELinuxCustom, SELinuxFacts, SELinuxRequestRPMs
-from leapp.libraries.stdlib import api, run, CalledProcessError
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import SELinuxCustom, SELinuxFacts, SELinuxModule, SELinuxModules, SELinuxRequestRPMs
 from leapp.reporting import Report
 
 TEST_MODULES = [
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/actor.py b/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/actor.py
index b764a51..23bc9c2 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/actor.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/actor.py
@@ -1,7 +1,7 @@
 from leapp.actors import Actor
-from leapp.models import SELinuxModules, SELinuxCustom, SELinuxFacts, SELinuxRequestRPMs, RpmTransactionTasks
-from leapp.tags import FactsPhaseTag, IPUWorkflowTag
 from leapp.libraries.actor import selinuxcontentscanner
+from leapp.models import RpmTransactionTasks, SELinuxCustom, SELinuxFacts, SELinuxModules, SELinuxRequestRPMs
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
 
 
 class SELinuxContentScanner(Actor):
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py b/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py
index e2f3de8..d512803 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py
@@ -2,8 +2,8 @@ import os
 import re
 from shutil import rmtree
 
-from leapp.libraries.stdlib import api, run, CalledProcessError
 from leapp.libraries.common.config import version
+from leapp.libraries.stdlib import api, CalledProcessError, run
 from leapp.models import SELinuxModule
 
 # types and attributes that where removed between RHEL 7 and 8
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/tests/component_test_selinuxcontentscanner.py b/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/tests/component_test_selinuxcontentscanner.py
index 2a98aa4..956277b 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/tests/component_test_selinuxcontentscanner.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/tests/component_test_selinuxcontentscanner.py
@@ -2,11 +2,11 @@ import os
 
 import pytest
 
-from leapp.snactor.fixture import current_actor_context
-from leapp.models import SELinuxModule, SELinuxModules, SELinuxCustom, SELinuxFacts, SELinuxRequestRPMs
 from leapp.libraries.common.config import mock_configs
-from leapp.libraries.stdlib import api, run, CalledProcessError
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import SELinuxCustom, SELinuxFacts, SELinuxModule, SELinuxModules, SELinuxRequestRPMs
 from leapp.reporting import Report
+from leapp.snactor.fixture import current_actor_context
 
 # compat module ensures compatibility with newer systems and is not part of testing
 TEST_MODULES = [
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxprepare/libraries/selinuxprepare.py b/repos/system_upgrade/common/actors/selinux/selinuxprepare/libraries/selinuxprepare.py
index 0ab2623..96e49b4 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxprepare/libraries/selinuxprepare.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxprepare/libraries/selinuxprepare.py
@@ -1,4 +1,4 @@
-from leapp.libraries.stdlib import api, run, CalledProcessError
+from leapp.libraries.stdlib import api, CalledProcessError, run
 from leapp.models import SELinuxModules
 
 
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/component_test_selinuxprepare.py b/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/component_test_selinuxprepare.py
index 078ee76..105dcb2 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/component_test_selinuxprepare.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/component_test_selinuxprepare.py
@@ -2,12 +2,11 @@ import os
 
 import pytest
 
-from leapp.libraries.stdlib import api, run, CalledProcessError
+from leapp.libraries.stdlib import api, CalledProcessError, run
 from leapp.models import SELinuxModule, SELinuxModules
 from leapp.reporting import Report
 from leapp.snactor.fixture import current_actor_context
 
-
 TEST_MODULES = [
     ["400", "mock1"],
     ["99", "mock1"],
diff --git a/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/unit_test_selinuxprepare.py b/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/unit_test_selinuxprepare.py
index beeaf3e..0a6f4c9 100644
--- a/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/unit_test_selinuxprepare.py
+++ b/repos/system_upgrade/common/actors/selinux/selinuxprepare/tests/unit_test_selinuxprepare.py
@@ -1,5 +1,5 @@
 from leapp.libraries.actor import selinuxprepare
-from leapp.libraries.stdlib import CalledProcessError, api, run
+from leapp.libraries.stdlib import api, CalledProcessError, run
 from leapp.models import SELinuxModule, SELinuxModules
 
 
diff --git a/repos/system_upgrade/common/models/selinux.py b/repos/system_upgrade/common/models/selinux.py
index 27c01ea..3305691 100644
--- a/repos/system_upgrade/common/models/selinux.py
+++ b/repos/system_upgrade/common/models/selinux.py
@@ -1,4 +1,4 @@
-from leapp.models import Model, fields
+from leapp.models import fields, Model
 from leapp.topics import SystemInfoTopic, TransactionTopic
```

File: repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py [hunk 4]
```diff
@@ -35,15 +39,42 @@ def process(self):
...
+        # get list of policy modules after the upgrade
+        installed_modules = set(
+            [x for (x, _) in selinuxapplycustom.list_selinux_modules()]
+        )
+
         # import custom SElinux modules
         for semodules in self.consume(SELinuxModules):
-            self.log.info("Processing custom SELinux policy modules. Count: %d.", len(semodules.modules))
+            self.log.info(
+                "Processing custom SELinux policy modules. Count: %d.",
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:50]: Could you use format here as well please?

File: repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py [hunk 5]
```diff
@@ -35,15 +39,42 @@ def process(self):
...
             for module in semodules.modules:
-                cil_filename = os.path.join(WORKING_DIRECTORY, "{}.cil".format(module.name))
-                self.log.info("Installing module %s on priority %d.", module.name, module.priority)
+                # check for presence of udica templates and make sure to install their latest versions
+                selinuxapplycustom.install_udica_templates(semodules.templates)
+                # Skip modules that are already installed. This prevents DSP modules installed with wrong
+                # priority (usually 400) from being overwritten by an older version
+                if module.name in installed_modules:
+                    self.log.info(
+                        "Skipping module %s on priority %d because it is already installed.",
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:60]: Please use format.

File: repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py [hunk 6]
```diff
@@ -35,15 +39,42 @@ def process(self):
...
+                        module.priority,
+                    )
+                    continue
+
+                # cil module files need to be extracted to disk in order to be installed
+                cil_filename = os.path.join(
+                    WORKING_DIRECTORY, "{}.cil".format(module.name)
+                )
+                self.log.info(
+                    "Installing module %s on priority %d.", module.name, module.priority
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:71]: Please use format

File: repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py [hunk 7]
```diff
@@ -35,15 +39,42 @@ def process(self):
...
+                )
                 if module.removed:
-                    self.log.warning("The following lines where removed because of incompatibility: \n%s",
-                                     '\n'.join(module.removed))
+                    self.log.warning(
+                        "The following lines where removed because of incompatibility: \n%s",
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:75]: Please use format function for string formatting, we have it in guidelines https://leapp.readthedocs.io/en/latest/python-coding-guidelines.html#prefer-new-style-string-formatting-to-old-style
(Won't mark other occurrencies not to be too annoying)

File: repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py [hunk 8]
```diff
@@ -107,34 +125,44 @@ def get_selinux_modules():
...
         try:
-            run(['semodule', '-c', '-X', priority, '-E', name])
+            run(["semodule", "-c", "-X", priority, "-E", name])
             # check if the module contains invalid types and remove them if so
             removed = check_module(module_file)
 
             # get content of the module
             try:
-                with open(module_file, 'r') as cil_file:
+                with open(module_file, "r") as cil_file:
```
Comment (fernflower) [inline repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py:154]: Nit: 'r' can be omitted

**Review by fernflower — COMMENTED** [review]
Some nits inline (nothing serious, string formatting and imports reorder suggestion). LGTM otherwise, thanks for the contribution!

**Comment by vmojzis** [comment]
> @vmojzis What is the top commit hash of your local master?
@fernflower
33053d7fd5a5dce44a03af0598069576903014e9
Same as this PR (vmojzis:selinux_).
Creating a new local branch and resetting my local master to origin/master fixed it, thanks.

File: repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py [hunk 9]
```diff
@@ -35,15 +39,42 @@ def process(self):
             self.log.warning("Failed to create working directory! Aborting.")
             return
 
+        # get list of policy modules after the upgrade
+        installed_modules = set(
+            [x for (x, _) in selinuxapplycustom.list_selinux_modules()]
```
[the remark this answers is not in the document]
Comment (vmojzis) [reply repos/system_upgrade/common/actors/selinux/selinuxapplycustom/actor.py:44]: Yes, this is just shorter. Should I change it (i.e. is your version easier to read)?

**Comment by vmojzis** [comment]
@fernflower Thanks for the review, it should all be fixed now.
I'm just still not sure about the backup directory (see #749 (comment)).

**Comment by fernflower** [comment]
> @fernflower Thanks, I'll fix that. Meanwhile I've been working on the reporting and it occurred to me that it could be useful to save any policy modules that failed to be applied on the upgraded system. Is there some path that leapp uses for such things, or can I just use something like /tmp/selinux-leapp (printing the modules as part of the report is not an option since they can be quite long)?
I am afraid we don't have an official location for things like this yet. Also I am not sure about /tmp (will that actually survive reboot?). How long can the list be? We already have pretty big things in report.

**Comment by fernflower** [comment]
@vmojzis We have a hm feature that not all report fields are shown in leapp-report.txt, while a full json dump is stored in leapp-report.json. What about adding failed modules as RelatedResources? 
I prefer putting them in the report file to some custom one, as leapp takes care of proper cleanup in between runs, while having an unmanaged custom one can lead to possible confusion.

**Comment by vmojzis** [comment]
> > @fernflower Thanks, I'll fix that. Meanwhile I've been working on the reporting and it occurred to me that it could be useful to save any policy modules that failed to be applied on the upgraded system. Is there some path that leapp uses for such things, or can I just use something like /tmp/selinux-leapp (printing the modules as part of the report is not an option since they can be quite long)?
> 
> I am afraid we don't have an official location for things like this yet. Also I am not sure about /tmp (will that actually survive reboot?). How long can the list be? We already have pretty big things in report.

Yes, /tmp is there just as a placeholder. Normally there shouldn't be any failed modules. The only reason for a failed module installation should be some complicated interdependency between the custom modules (or required custom modules installed on priority 100 -- reserved for distribution policy modules). Both cases should be very rare. On the other hand, there is no limit to how many custom modules or "semanage" commands a system can have. Each module usually has hundreds of lines and needs to be in the form of a file to be installed -- hence the preference to save them as files.

> We have a hmm feature that not all report fields are shown in leapp-report.txt, while a full json dump is stored in leapp-report.json. What about adding failed modules as RelatedResources?

I'm actually already doing that. Does that include their content in the report?

**Comment by fernflower** [comment]
> I'm actually already doing that. Does that include their content in the report?
> https://github.com/oamg/leapp-repository/pull/749/files#diff-7b3410d4a2f5284d190fcacc1cd90255b792a10baef32d37629158586277ffccR167-R170

Related resource will be shown in leapp-report.json only, not leapp-report.txt.
Got your point about the need to store failed modules to install, will create a leapp issue about custom backup dir.

**Comment by fernflower** [comment]
leapp-ci build

**Comment by fernflower** [comment]
leapp ci-build

**Comment by vmojzis** [comment]
I just squashed the required packages in `selinuxapplycustom/Makefile` to a single line -- should make test runs faster (requested by @pirat89). No other changes.

File: repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py [hunk 10]
```diff
@@ -2,18 +2,28 @@
...
+                     "ganesha_var_log_t", "ganesha_var_run_t", "ganesha_use_fusefs"]
+
+# types and attributes that where removed between RHEL 8 and 9
+REMOVED_TYPES_EL8 = ['cephfs_t', 'cgdcbxd_exec_t', 'cgdcbxd_t', 'cgdcbxd_unit_file_t', 'cgdcbxd_var_run_t',
+                     'cloud_what_var_cache_t', 'journal_remote_client_packet_t', 'journal_remote_port_t',
+                     'journal_remote_server_packet_t', 'kdbusfs_t', 'logging_syslogd_list_non_security_dirs',
+                     'nvme_device_t', 'pcp_pmmgr_exec_t', 'pcp_pmmgr_initrc_exec_t', 'pcp_pmmgr_t',
+                     'pcp_pmwebd_exec_t', 'pcp_pmwebd_initrc_exec_t', 'pcp_pmwebd_t', 'rpm_transition_domain',
+                     'systemd_journal_upload_exec_t', 'systemd_journal_upload_t',
+                     'systemd_journal_upload_var_lib_t', 'virt_qmf_exec_t', 'virt_qmf_t']
```
Comment (pirat89) [inline repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py:26]: mix of double&single quotes

**Comment by pirat89** [comment]
* descriptions of actors should be updated

File: repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py (hunk 10 above)
Comment (vmojzis) [reply repos/system_upgrade/common/actors/selinux/selinuxcontentscanner/libraries/selinuxcontentscanner.py]: Thanks, fixed.

**Comment by vmojzis** [comment]
> * descriptions of actors should be updated
> 
> Instead of e.g. RHEL 7, it could refer to "the original RHEL system", Instead of RHEL 8 / 9, it could refer to "the target RHEL system" or "the upgraded RHEL system". Depends on context of messages. As the actors are used for IPU 7 -> 8 and 8 -> 9, description should be clear

Right, thanks, fixed.

**Comment by vmojzis** [comment]
Rebased as per Petr Stodulka's email. 
Ad. IntegrationTestsMissing label: @Rezney should I add the integration tests (If so, could you please point me to some example from other actors)?

**Comment by fernflower** [comment]
@vmojzis lgtm, regression tests are also fine. However our integration tests are not that specific so they don't cover your specific usecase, these should be tested by the relevant SST. We can merge it though, it looks mature enough, thank you!

**Review by fernflower — APPROVED** [review]
Lgtm. Could you please do one last rebase?

**Comment by vmojzis** [comment]
> @vmojzis lgtm, regression tests are also fine. However our integration tests are pretty generic so they don't cover your specific usecases, these should be tested by the relevant SST. We can merge it though, it looks mature enough, thank you!

We have some integration test of our own:
https://src.osci.redhat.com/tests/selinux-policy/blob/master/f/Sanity/inplace-upgrade
https://docs.engineering.redhat.com/display/SELINUX/System+upgrades+testing

**Comment by fernflower** [comment]
@vmojzis Hah, looks like we are already running your selinux-policy tests as part of our regression suite =) Btw it passes both for 7to8 and 8to9, you can take a look at the results if interested.

**Pull request merged**